rust/src/types/opcode.rs

```diff
@@ -0,0 +1,364 @@
+#![allow(dead_code)]
+
+const STOP: u8 = 0x00;
```

Comment:
Also a curiosity question: why are there `u8` constants for the opcodes as well as the unit-only enum constants of type `Opcode`? Is this to ease the definition of test cases because with the Opcodes-only you would have to write

```
[Opcode:ADD as u8, Opcode:PUSH1 as u8, 4, Opcode:STOP as u8]
```
and with these constants you can write
```
[ADD, PUSH1, 4, STOP]
```
or is there another use case for this?

Maybe the latter is also not possible, because it does not seem to be used in test cases ...

Reply:
Ideally I would like to have only the `Opcode` enum and convert to `u8` when needed. The problem is that in the match in `code_byte_type` I have to match against `u8` but type casts are not possible in match arms. The tests in the `code_state` module use `Opcode` instead of the consts because the consts are private to the `opcode` module whereas `Opcode` is public. I want to keep the consts private because that makes it easier to get rid if them at a later point if possible.

Reply:
BTW
I just tried to figure out if it is possible to avoid the consts completely.
```rs
match code_byte {
    ...
    Opcode::Push9 as u8 => (CodeByteType::Opcode, 9), // preferred but not possible
    v if v == Opcode::Push9 as u8 => (CodeByteType::Opcode, 9), // possible but ugly
    const { Opcode::Push9 as u8 } => (CodeByteType::Opcode, 9), // possible on nightly
    ...
}
```